libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/RequireMapboxNavigationDelegate.kt

```diff
@@ -0,0 +1,135 @@
+ * a couple examples of how you may use the delegate.
+ *
+ * Default can be used when [MapboxNavigationApp] is setup elsewhere.
+ * ```
+ * val mapboxNavigation by requireMapboxNavigation()
+ * ```
+ *
+ * Initialize the [MapboxNavigationApp] when you are ready to use it
+ * ```
+ * val mapboxNavigation by requireMapboxNavigation {
+ *   MapboxNavigationApp.setup(..)
+ * }
+ * ```
+ *
+ * Register subscriptions and setup MapboxNavigationApp
+ * ```
+ * private val mapboxNavigation by requireMapboxNavigation(
+ *   onResumedObserver = object : MapboxNavigationObserver {
+ *     override fun onAttached(mapboxNavigation: MapboxNavigation) {
+ *       mapboxNavigation.registerLocationObserver(locationObserver)
+ *       mapboxNavigation.registerRoutesObserver(routesObserver)
+ *     }
+ *     override fun onDetached(mapboxNavigation: MapboxNavigation) {
+ *       mapboxNavigation.unregisterLocationObserver(locationObserver)
+ *       mapboxNavigation.unregisterRoutesObserver(routesObserver)
+ *     }
+ *   }
+ * ) {
+ *   MapboxNavigationApp.setup(
+ *       NavigationOptions.Builder(this)
+ *           .accessToken(accessToken)
+ *           .build()
+ *     )
+ * }
+ * ```
+ *
+ * @see [RequireMapboxNavigationDelegate] for more details.
+ */
+fun requireMapboxNavigation(
+    onCreatedObserver: MapboxNavigationObserver? = null,
+    onStartedObserver: MapboxNavigationObserver? = null,
+    onResumedObserver: MapboxNavigationObserver? = null,
+    onInitialize: (() -> Unit)? = null,
+) = RequireMapboxNavigationDelegate(
+    onCreatedObserver = onCreatedObserver,
+    onStartedObserver = onStartedObserver,
+    onResumedObserver = onResumedObserver,
+    onInitialize = onInitialize
+)
+
+/**
+ * Attaches a [LifecycleOwner] to [MapboxNavigationApp] and provides access to [MapboxNavigation].
+ *
+ * You can choose to call [MapboxNavigationApp.setup] in the [onInitialize]. You can also setup in
+ * the onCreate calls, or any call that happens before this delegate is accessed. The delegate will
+ * crash if the app is not setup and an attached has been created.
+ *
+ * You can use the observers parameter to setup any subscriptions. This is important because the
+ * [MapboxNavigation] instance can be re-created with [MapboxNavigationApp.disable], or if all
+ * [MapboxNavigationApp.attach] lifecycles are destroyed.
+ *
+ * @param onCreatedObserver registered to the [Lifecycle.State.CREATED] lifecycle
+ * @param onStartedObserver registered to the [Lifecycle.State.STARTED] lifecycle
+ * @param onResumedObserver registered to the [Lifecycle.State.RESUMED] lifecycle
+ * @param onInitialize called when the property is read for the first time
+ */
+class RequireMapboxNavigationDelegate(
+    private val onCreatedObserver: MapboxNavigationObserver? = null,
+    private val onStartedObserver: MapboxNavigationObserver? = null,
+    private val onResumedObserver: MapboxNavigationObserver? = null,
+    private val onInitialize: (() -> Unit)? = null
```

Comment:
could convert this into a builder style so it has better protections from SEMVER

https://github.com/mapbox/mapbox-maps-android/blob/main/extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMapScreenInstaller.kt

Reply:
Ok made a fix here. The delegate class is internal so we could change the implementation or details.

Now the extension function is, this is better suited for the stable api
```
fun requireMapboxNavigation(
    onCreatedObserver: MapboxNavigationObserver? = null,
    onStartedObserver: MapboxNavigationObserver? = null,
    onResumedObserver: MapboxNavigationObserver? = null,
    onInitialize: (() -> Unit)? = null,
): ReadOnlyProperty<LifecycleOwner, MapboxNavigation>
```